### DISCUSSION TITLE
Docker Image - Expected life cycle

### DISCUSSION THREAD

**Discussion by george0st** [mlrun/mlrun#1936] [Q&A] [answered]
Hi @Tankilevitch, are there any rules when you create a new docker image? E.g. always after the release of the major version or after the release of the first revision for major release, etc.? Many thanks for feedback.

**Comment by Tankilevitch** [comment, marked as answer]
Hello, @george0st, Which Docker images are you referring to?

**Reply by george0st** [reply]
Hi @Tankilevitch, I think about MLRun (mlrun-kit) image.

**Reply by Tankilevitch** [reply]
Hi @george0st,
As part of the release process For each MLRun release we also upload all the related images to ( `https://hub.docker.com/search?q=mlrun` and `quay.io/mlrun/<image_name>`).
Currently the kit doesn't get updated to the newest MLRun version.
But there is another option ( which I'll add also to the installations docs) to use newer MLRun images.
You can update your existing release by running :
```
helm --namespace mlrun \
   upgrade mlrun-kit \
   --reuse-values \ 
   --set mlrun.ui.image.tag=1.0.0 \ 
   --set mlrun.api.image.tag=1.0.0 \
   --set jupyterNotebook.image.tag=1.0.0 \
   v3io-stable/mlrun-kit
```
Or you can remove the current one and start from the beginning, follow [this](https://docs.mlrun.org/en/latest/install.html#uninstalling-the-chart) to uninstall, and run the following in your helm installation step:
```
helm --namespace mlrun \
   install mlrun-kit \
   --set mlrun.ui.image.tag=1.0.0 \ 
   --set mlrun.api.image.tag=1.0.0 \
   --set jupyterNotebook.image.tag=1.0.0 \
   v3io-stable/mlrun-kit
```
I'll suggest to use only official releases and not pre-releases.

**Reply by Tankilevitch** [reply]
You can also update to newer Nuclio versions with the following configurations : 
```
--set nuclio.controller.image.tag=<nuclio-image-version>
--set nuclio.dashboard.image.tag=<nuclio-image-version>
```
If I recall correctly nuclio dropped support for k8s versions lower than 1.22 from nuclio 1.7.5, so take this in mind.

**Reply by george0st** [reply]
@Tankilevitch, it is very useful information and I will use it, really many thanks.
